### ISSUE TITLE
Improve the continuous integration workflow (CI)

### ISSUE THREAD

**Issue by vladimiry** [vladimiry/ElectronMail#133]
Currently, Travis CI is used for preparing Linux and macOS packages and AppVeyor for Windows packages, so the process is scattered over those CI systems. There is a job that builds built-in packages from the sources and it needs to be executed on Linux/macOS systems as neither Protonmail nor the Tutanota provides flawless UX for building their code on Windows.

I'd like to go with a single CI system that would support Linux, macOS and Windows systems so I could use a single CI config file. This would simplify the CI config file and packages preparing process and would make it easier for other developers to independently build the packages from the provided source code.

Besides would like CI supports sharing artefacts between the jobs, so I could prepare the built-in web clients executing the top-level Linux job and then 3 dependent separate jobs for Linux/macOS/Windows would pick up the prepared web clients as an artefact. This would speed up the process. I currently upload the web clients artefact to hidden Github release since Travis CI doesn't support sharing artefacts between the jobs.

I'm going to start exploring the options from Gitlab CI.

**Comment by vladimiry** [comment]
I ended up using Travis CI for building on all the platforms, so AppVeyor becoming obsolete. Travis CI supports Windows, it is still in beta stage though. I've not yet figured how to make e2e tests working there on Windows, but building from source code and preparing the installation packages works.

I also decided to stop automatic submitting installation packages from CI to Github release as this feature requires GH_TOKEN to be imported into the CI environment. That can potentially lead to the GH_TOKEN exposure which would a severe security issue as that token allows extended access to a respective GitHub account. Something like that actually [happened just recently with Docker Hub](https://success.docker.com/article/docker-hub-user-notification).

Disabling automatic installation packages publishing will also lead to disabling automatic update feature as it depends on automatic publishing. But this is not a big deal since it worked only for Windows packages (for other systems it requires packages to be signed).

Besides that, I'm going to publish electron@v4-based releases with only AppImage / Snap packages included in v3.2.100+ versions range. So the electron@v5-based releases won't include the AppImage / Snap packages and will be published in v3.3+.x versions range. It's because there is still unresolved issue of running electron@v5-based AppImage / Snap packages on Linux with only [SUID Sandbox](https://chromium.googlesource.com/chromium/src/+/HEAD/docs/linux_suid_sandbox_development.md) available (means no [User Namespaces](https://wiki.archlinux.org/index.php/Linux_Containers) feature enabled). Related electron issue is [here](https://github.com/electron/electron/issues/17972), but I believe it's something that should be solved with AppImage / Snap maintainers involved. All the releases are going to be published as pre-release until the issue is solved.
